### PR TITLE
gpu based examples update to add torchmetrics

### PR DESCRIPTION
**Pull request by iArpanPatel**
To add torchmetrics for gpu based examples user image build files.

### CODE REVIEW

File: examples/mnist-pyt/gpu-based/nvidia/swci/taskdefs/build_pyt_user_image.yaml [hunk 1]
```diff
@@ -15,4 +15,5 @@ Body:
     - RUN mkdir -p /tmp/hpe-swarmcli-pkg
     - COPY swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl  /tmp/hpe-swarmcli-pkg/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl
     - RUN pip install /tmp/hpe-swarmcli-pkg/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl
+    - RUN pip3 install torchmetrics
```
Comment (girisairam450) [inline examples/mnist-pyt/gpu-based/nvidia/swci/taskdefs/build_pyt_user_image.yaml:18]: why pip3 why not pip ? Can we keep it consistent

File: examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml [hunk 2]
```diff
@@ -16,4 +16,5 @@ Body:
     - COPY swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl  /tmp/hpe-swarmcli-pkg/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl
     - RUN pip install /tmp/hpe-swarmcli-pkg/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl
     - RUN pip install pyamdgpuinfo==2.1.2
+    - RUN pip3 install torchmetrics
```
Comment (girisairam450) [inline examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml:19]: why pip3 why not pip ? Can we keep it consistent

**Review by girisairam450 — CHANGES_REQUESTED** [review]
Added review comment

File: examples/mnist-pyt/gpu-based/nvidia/swci/taskdefs/build_pyt_user_image.yaml (hunk 1 above)
Comment (iArpanPatel) [reply examples/mnist-pyt/gpu-based/nvidia/swci/taskdefs/build_pyt_user_image.yaml]: Corrected

File: examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml (hunk 2 above)
Comment (iArpanPatel) [reply examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml]: Corrected

File: examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml [hunk 3]
```diff
@@ -16,4 +16,5 @@ Body:
     - COPY swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl  /tmp/hpe-swarmcli-pkg/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl
     - RUN pip install /tmp/hpe-swarmcli-pkg/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl
     - RUN pip install pyamdgpuinfo==2.1.2
+    - RUN pip install torchmetrics
```
Comment (htjain) [inline examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml:19]: merge with above `pip install` .
like `pyamdgpuinfo` why are not fixing version number for `torchmetrics`

**Review by RadhakrishnaJ — APPROVED** [review]

File: examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml (hunk 3 above)
Comment (girisairam450) [reply examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml:19]: Don't have to freeze torchmetrics version as it is picking different versions in cpu and gpu examples

Comment (girisairam450) [reply examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml:19]: Else we have to freeze it as torchmetrics==0.11.4 (for cpu) and torchmetrics==1.2.0 (for gpu)

**Review by girisairam450 — APPROVED** [review]
changes are okay

File: examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml (hunk 3 above)
Comment (iArpanPatel) [reply examples/mnist-pyt/gpu-based/amd/swci/taskdefs/build_pyt_user_image.yaml:19]: We will let the torchmetrics version get selected according to base image.

**Pull request merged**